### PR TITLE
Add method to Mailchimp for signing up or updating a user for a nonprofit

### PR DESCRIPTION
**Pull request by camckin10**
Closes  https://github.com/CommitChange/tix/issues/4105

We need to add an method to app/legacy_lib/mailchimp.rb for signing up and updating a user for a nonprofit. 
The method should be a class method on Mailchimp called signup_nonprofit_user. 
The method should work very similar to Mailchimp.signup.
We should test this by mocking the put call to Mailchimp in spec/lib/mailchimp_spec.rb.

**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

### CODE REVIEW

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: Why was this deleted?

File: app/legacy_lib/mailchimp.rb
Comment (wwahammy) [inline app/legacy_lib/mailchimp.rb]: You should be using this one because it's a "class method". Class methods are defined when you add `self.` before the method name in the definition. (There are other ways but for our purposes, that's enough detail)

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: Hold off on this until you finish the previous three tasks in https://github.com/CommitChange/tix/issues/4094

**Review by wwahammy — CHANGES_REQUESTED** [review]

File: spec/lib/mailchimp_spec.rb
Comment (camckin10) [reply spec/lib/mailchimp_spec.rb]: Must be Github because that change does not appear locally.

Comment (wwahammy) [reply spec/lib/mailchimp_spec.rb]: Perhaps but it needs to be corrected :smile:

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb:80]: Please remove these changes.

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb:98]: Were we previously NOT testing this function before this review?

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: Why was `user2` added? It doesn't seem to be used in the test.

File: spec/lib/mailchimp_spec.rb
Comment (wwahammy) [inline spec/lib/mailchimp_spec.rb]: Please remove

File: app/legacy_lib/mailchimp.rb
Comment (wwahammy) [inline app/legacy_lib/mailchimp.rb:55]: You'll need a spec to verify that it works. You will need to mock the `put` call. I can pair with you on figuring out the best way to do that.

**Review by wwahammy — CHANGES_REQUESTED** [review]
Thanks @camckin10. I had a few questions and changes I'd like made. Specifically,I'd like to mock the put call to Mailchimp's api before merging. We should also manually test that it works against Mailchimp's real API.

File: spec/lib/mailchimp_spec.rb
Comment (camckin10) [reply spec/lib/mailchimp_spec.rb:98]: Yes, however, that was because the ticket was out of order. 
Also, the test cases that I had at that time did not run.

File: spec/lib/mailchimp_spec.rb
Comment (camckin10) [reply spec/lib/mailchimp_spec.rb:80]: What changes are you asking to be removed? @wwahammy 
We found that those lines had been removed so I am adding them back.

Comment (wwahammy) [reply spec/lib/mailchimp_spec.rb:80]: The white space is different from the original, which means the lines are different for Git. It's good practice to avoid having any changes in a file unless they relate to the actual changes being considered.

**Review by wwahammy — APPROVED** [review]
Looks good, we can merge and I'll create a new ticket for creating the test.

**Pull request merged**
